07-http_gordle/final/internal/gordle/corpus.go

```diff
@@ -0,0 +1,45 @@
+package gordle
+
+import (
+	"fmt"
+	"math/rand"
+	"os"
+	"strings"
+	"time"
+)
+
+const (
+	// ErrInaccessibleCorpus is returned when the corpus can't be loaded.
+	ErrInaccessibleCorpus = corpusError("corpus can't be opened")
+	// ErrEmptyCorpus is returned when the provided corpus is empty.
+	ErrEmptyCorpus = corpusError("corpus is empty")
+)
+
+// ReadCorpus reads the file located at the given path
+// and returns a list of words. If it fails, the error is ErrInaccessibleCorpus or ErrEmptyCorpus.
+func ReadCorpus(path string) ([]string, error) {
+	data, err := os.ReadFile(path)
+	if err != nil {
+		return nil, fmt.Errorf("unable to open %q for reading (%s): %w", path, err, ErrInaccessibleCorpus)
+	}
+
+	if len(data) == 0 {
+		return nil, ErrEmptyCorpus
+	}
+
+	// we expect the corpus to be a line- or space-separated list of words
+	words := strings.Fields(string(data))
```

Comment:
I would verify that the words match what you expect. e.g. can they contain `=` and whether `[]rune(` conversion is valid for those words.